Realm/RLMRealm+Sync.mm

```diff
@@ -0,0 +1,53 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMRealm+Sync.h"
+
+#import "RLMObjectSchema.h"
+#import "RLMRealm_Private.hpp"
+#import "RLMResults_Private.hpp"
+#import "RLMSchema.h"
+
+#import "results.hpp"
+#import "sync/partial_sync.hpp"
+#import "shared_realm.hpp"
+
+using namespace realm;
+
+@implementation RLMRealm (Sync)
+
+- (void)fetchResultsForQuery:(NSString *)query objectType:(Class)type callback:(RLMPartialSyncFetchCallback)callback {
+    NSString *className = NSStringFromClass(type);
```

Comment:
Needs to check if the Realm is actually a partially synced Realm, and throw/return early if not.

Reply:
I'd suggest pushing that checking down to object store if possible.